go-controller/pkg/ovn/gateway_init.go

```diff
@@ -216,7 +223,9 @@ func (oc *Controller) gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet,
 		p := func(item *nbdb.LogicalRouterStaticRoute) bool {
 			return item.Nexthop == lrsr.Nexthop && item.IPPrefix == lrsr.IPPrefix
 		}
-		err := libovsdbops.CreateOrUpdateLogicalRouterStaticRoutesWithPredicate(oc.nbClient, types.OVNClusterRouter, &lrsr, p)
+		// TODO(trozet): evaluate if all fields can be updated
+		err := libovsdbops.CreateOrUpdateLogicalRouterStaticRoutesWithPredicate(oc.nbClient, types.OVNClusterRouter,
+			&lrsr, p, &lrsr.Nexthop, &lrsr.IPPrefix)
```

Comment:
done

Reply:
hmm now this is behaving as `onModelUpdatesAllNonDefault` which is fine by me, but there is the ambiguity that it could behave as `onModelUpdatesNone` and it would be equally fine.

This construct you used inside 
```
if len(fields) == 0 {
		fields = onModelUpdatesAllNonDefault()
	}
```
does not give the option from the outside to specify the behavior, which is also fine by me as if we would care I would prefer to have specific `Create` methods we already have where we do care.

But subtleties....